packages/extension-url-mapping/src/setup-app.ts

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+// Third-Party
+import React from 'react'
+import {RouteProps} from 'react-router-dom'
+
+// Platform Imports
+import {
+    ApplicationExtension,
+    SliceInitializer,
+    withApplicationExtensionStore
+} from '@salesforce/pwa-kit-extension-sdk/react'
+import {applyHOCs} from '@salesforce/pwa-kit-extension-sdk/react/utils'
+import {GetRoutesParams, BeforeRouteMatchParams} from '@salesforce/pwa-kit-extension-sdk/types'
+import Auth from '@salesforce/commerce-sdk-react/auth'
+import {ShopperSeo} from 'commerce-sdk-isomorphic'
+import {routeComponent} from '@salesforce/pwa-kit-react-sdk/ssr/universal/components/route-component'
+
+// Local Imports
+import {Config} from './types'
+import SamplePage from './pages/sample'
+
+// Overridable Imports
+// Using the `overridable` loader means that you are opting in to the override module resolution flow. As a result this module
+// will be resolved by first looking in the base projects `overrides` folder then the overrides folders of any extensions configured
+// after this one. Only if no module is found will the referenced module in this project be used.
+import sampleHOC from 'overridable!./components/sample-hoc'
+
+// Others
+import extensionMeta from '../extension-meta.json'
+
+interface StoreSlice {
+    count: number
+    increment: () => void
+    decrement: () => void
+}
+
+// This is safe to delete if your extension does not use state. If you aren't using this, ensure you remove the
+// `withApplicationExtensionStore` usage below as well.
+const sliceInitializer: SliceInitializer<StoreSlice> = (set) => ({
+    count: 0,
+    increment: () => set((state) => ({count: state.count + 1})),
+    decrement: () => set((state) => ({count: state.count - 1}))
+})
+
+class UrlMapping extends ApplicationExtension<Config> {
```

Comment:
I created a new application extension to get a feel of what it's like to build a url-mapping extension. I'll remove it before merging the PR.